pkg/volume/emptydir/empty_dir.go

```diff
@@ -301,12 +302,17 @@ func (ed *emptyDir) assignQuota(dir string, mounterSize *resource.Quantity) erro
 		if err != nil {
 			klog.V(3).Infof("Unable to check for quota support on %s: %s", dir, err.Error())
 		} else if hasQuotas {
-			klog.V(4).Infof("emptydir trying to assign quota %v on %s", mounterSize, dir)
-			err := fsquota.AssignQuota(ed.mounter, dir, ed.pod.UID, mounterSize)
-			if err != nil {
-				klog.V(3).Infof("Set quota on %s failed %s", dir, err.Error())
+			projid, err := fsquota.GetQuotaOnDir(ed.mounter, dir)
+			if err == nil && projid != common.BadQuotaID {
+				klog.V(4).Infof("emptydir trying to assign quota %v on %s", mounterSize, dir)
+				err := fsquota.AssignQuota(ed.mounter, dir, ed.pod.UID, mounterSize)
+				if err != nil {
+					klog.V(3).Infof("Set quota on %s failed %s", dir, err.Error())
+				}
+				return err
+			} else {
+				klog.V(4).Infof("Attempt to check quota on dir %s failed: %v", dir, err)
```

Comment:
When `err == nil && projid == common.BadQuotaID`, this will log "`... failed: <nil>`", which is odd. In what cases can this happen and what does it mean to an user?

Reply:
This seems to be impossible.
- `BadQuotaID QuotaID = 0`
<img width="979" alt="image" src="https://user-images.githubusercontent.com/2010320/210794185-d72c4d82-0fbf-4b6a-ab0e-06e766464486.png">

```
if err == nil && projid != common.BadQuotaID {
```
can change to 
```
if err == nil {
```

Should I change it?

Reply:
Yes please, it would be more obvious.

Reply:
Updated.